src/App.tsx

```diff
@@ -9,39 +9,30 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { fetchUsers } from './features/userSlice';
+import { fetchPosts, postsSelector } from './features/postSlice';
+import { authorSelector } from './features/authorSlice';
+import { selectedPostSelector } from './features/selectedPostSlice';
+import { set } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const { items: posts, loaded, hasError } = useAppSelector(postsSelector);
+  const author = useAppSelector(authorSelector);
+  const selectedPost = useAppSelector(selectedPostSelector);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  useEffect(() => {
+    dispatch(fetchUsers());
+  }, [dispatch]);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(set(null));
 
     if (author) {
-      loadUserPosts(author.id);
-    } else {
-      setPosts([]);
+      dispatch(fetchPosts(author.id));
     }
-  }, [author]);
+  }, [author, dispatch]);
```

Comment:
The `useEffect` hook here is set to dispatch `fetchPosts` when the `author` changes. Ensure that the `author` is correctly set in the state using the `set` action from `authorSlice`. Also, verify that `fetchPosts` is correctly implemented to fetch posts based on the author's ID .